### ISSUE TITLE
Standard set of test raw blocks and transactions that stress all edge cases

### ISSUE THREAD

**Issue by AustEcon** [electrumsv/electrumsv-sdk#6]
I have a fair idea of most of the pitfalls that need to be handled wrt indexing the chain. 

Things like handling of malformed output scripts (which can legitimately happen where they do not comply with bitcoin script syntax rules). Or coinbases which do not get validated so can just find total junk in there...

But also the "happy path" of correctly parsing all pushdatas / opcodes.

So I think a set of hand-crafted raw blocks and raw transactions with expected output would be nice (but might be a bit tedious to make...)

**Comment by rt121212121** [comment]
I wonder if it would be best to generate these and then you can rebuild the set.

**Comment by AustEcon** [comment]
Yeah that's probably the easiest way. Just generate a block on RegTest densely packed with all of the trouble-maker transactions perhaps and then gather up all the utxos... the full state of the chain and capture it for testing?

I don't think you can make custom coin bases when mining blocks but for everything else would be ideal.

**Comment by rt121212121** [comment]
I think we need to be able to make custom coin bases if we want to test miner id, but actually emulating that might be outside the scope of our sdk.
